### PR TITLE
Parallel Identification Reading and Relaxed file naming in input psms list

### PR DESCRIPTION
**Pull request by trishorts**
Input processing is not currently parallel. IDs are read line by line and processed individually. This PR reads all the input lines, breaks them into groups by file, then processes all the ids in that file before adding them to the group. 

In this PR, we also allow flexibility in filenames for the psms input file. The filename can be with or without extension and can have or have not a full file path. also windows and linux filepaths are explicitly tested.

### CODE REVIEW

**Review by Alexander-Sol — COMMENTED** [review]
The relaxed file naming looks fine, but I'm not seeing any changes related to parallel identification reading. Am I missing something?

**Comment by Alexander-Sol** [comment]
> The relaxed file naming looks fine, but I'm not seeing any changes related to parallel identification reading. Am I missing something?

I was missing something. Got it

**Comment by Alexander-Sol** [comment]
Test coverage looks good, but I don't have a linux machine to be absolutely sure.

**Review by Alexander-Sol — APPROVED** [review]

File: Util/PsmReader.cs [hunk 1]
```diff
@@ -81,304 +81,464 @@ public static List<Identification> ReadPsms(string filepath, bool silent, List<S
...
+                return null;
+            }
+
+            // retention time
+            double ms2RetentionTime = -1;
+            //percolator input files do not have retention times. So, we have to get them from the data file using the scan number.
+
+            if (int.TryParse(param[_msmsScanCol], NumberStyles.Number, CultureInfo.InvariantCulture, out int scanNumber))
+            {
+                ms2RetentionTime = scanHeaderInfo.Where(i => Path.GetFileNameWithoutExtension(i.FileNameWithoutExtension) == Path.GetFileNameWithoutExtension(fileName) && i.ScanNumber == scanNumber).FirstOrDefault().RetentionTime;
```
Comment (rmillikin) [inline Util/PsmReader.cs:448]: this could be made faster with a dict lookup instead of a list but it's probably fine for now

**Pull request merged**